src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -2558,7 +2566,10 @@ _public_ int sd_varlink_error(sd_varlink *v, const char *error_id, sd_json_varia
         } else
                 varlink_set_state(v, VARLINK_PROCESSED_METHOD);
 
-        return 1;
+        /* Everything worked. Let's now return the error we got passed as input as negative errno, so that
+         * programs can just do "return sd_varlink_error();" and get both: a friendly error reply to clients,
+         * and an error return from the current stack frame. */
+        return sd_varlink_error_to_errno(error_id, parameters);
```

Comment:
Looks like at least resolved produces a lot of spurious "Invalid request descriptor" error messages on errors now
```sh
Found synthetic NXDOMAIN response.
varlink-27-27: Sending message: {"error":"io.systemd.Resolve.DNSError","parameters":{"rcode":3}}
varlink-27-27: Changing state processing-method → processed-method
Failed to send address reply: Invalid request descriptor
varlink-27-27: Connection busy.
varlink-27-27: Changing state processed-method → idle-server
varlink-27-27: Got POLLHUP from socket.
varlink-27-27: Changing state idle-server → pending-disconnect
```

Reply:
Thanks. Fix is waiting in #34962.